feat: add ssao settings

BrianAmadori opened the pull request:
## What? 

* A new `Ultra` profile was added on top of the `High` profile.
* All profiles were modified.
* Add new setting spinbox for the SSAO feature.
* Fixed potential bug related to float point precision that could cause the preset spinbox to skip levels and go to `custom` randomly.
* Upped the "Match 720p" resolution a bit to be more decent.

AjimenezDCL committed:
Project Ported

AjimenezDCL committed:
AO Compatibility with our shaders

AjimenezDCL committed:
Unity 2020.2b hacked into kernel

AjimenezDCL committed:
Now our DCL Lit shader works with SSAO in webgl

AjimenezDCL committed:
Remove WebGL postprocess to avoid the custom DCL template

AjimenezDCL committed:
fixing lint

AjimenezDCL committed:
fixing lint + disable unity from CI

AjimenezDCL committed:
disable unity from CI

AjimenezDCL committed:
disable unity from CI

AjimenezDCL committed:
$%@#$ lint

AjimenezDCL committed:
Avoid replacing the built-in unity build

AjimenezDCL committed:
Avoid replacing the built-in unity build

AjimenezDCL committed:
Fix some CI issues

AjimenezDCL committed:
More CI issues

AjimenezDCL committed:
More CI issues!

AjimenezDCL committed:
More CI issues!

AjimenezDCL committed:
Removed issue with mouse position

AjimenezDCL committed:
Tweaked SSAO

AjimenezDCL committed:
Tweaked SSAO

AjimenezDCL committed:
Fixing mouse mispositioning: attempt 1

AjimenezDCL committed:
Compiled Toon Shader for URP 10

AjimenezDCL committed:
Added DepthNormal pass and SSAO to toon shader

AjimenezDCL committed:
Tentative fix for avatar editor missing.

AjimenezDCL committed:
Updated to Unity 2020.2.0f1

Commit:
merge with master

Commit:
fix: resolution change not working on Unity 2020

Commit:
fix: pedantic linter fail

Commit:
chore: version update to 2020.3.0

Commit:
fix: index.html points to the proper .unityweb files + new build upload

Commit:
try to cache bust

Commit:
CI for 2020.3.0

Commit:
remove hardcoded build

Commit:
MainTest scene double event system issue

Commit:
MainTest lighting file

Commit:
SSAO tweaks

Commit:
backpack visual fix

Commit:
backpack visual fix (test 2)

Commit:
change library cache id

Commit:
revert test

Commit:
shader test

Commit:
update gitignore

Commit:
updated srp packages

Commit:
make custom SSAO copy for DCL bugfixes

Commit:
fix script name in ForwardRenderer

Commit:
meta fix

Commit:
test

Commit:
ssao patch 2

Commit:
ssao patch 3

Commit:
ssao patch 4

Commit:
test 5

Commit:
fix: displaced SSAO in avatar preview

- also disabled cameras when avatar editor is opened to improve performance

Commit:
SSAO tweaks

Commit:
proper porting of Lit and Unlit shaders

- matched implementation of latest URP version
- bloom tweaks to match old version the best as possible, had to hardcode a * 5 in the color emission value and change bloom values
- had to tweak SSAO values again because everything changed
- re-applied all our custom fixes for GLTF
- transparencies were fixed

Commit:
remove override SSAO shader because is not needed anymore

Commit:
update materials

Commit:
AO tweaks

Commit:
Remade toon shader integration w/SSAO

Commit:
transparency patch

Commit:
fix: disable SSAO and DepthNormals pass for transparent materials

- updated prebuilt shaders and variant collections

Commit:
transparency fix

Commit:
feat: add ssao settings

Commit:
new preset first draft

Commit:
Merge branch 'master' into chore/unity-2020-3-0-f1

# Conflicts:
#	kernel/static/unity/Build/DCLUnityLoader.js
#	unity-client/Assets/ABConverter/Core.cs
#	unity-client/Assets/Scenes/Test/MainTest.unity
#	website/public/index.html

Commit:
kernel code cleanup

Commit:
Merge branch 'master' into chore/unity-2020-3-0-f1

# Conflicts:
#	website/scripts/hash_generator.js

Commit:
add SetSSAO method for debugging

Commit:
make SendMessageToUnity public to allow for advanced debugging

Commit:
fix shader issue

Commit:
culling controller test fix

Commit:
disable SSAO + replace SetSSAO with ToggleSSAO to avoid invalid payload issue (SendMessage doesn't accept lonely bools)

Commit:
code cleanup

Commit:
Merge branch 'master' into chore/unity-2020-3-0-f1

Commit:
fix tutorial alice materials

Commit:
add alpha clip to shadervariants and prewarm assets

Commit:
update sprite atlases

Commit:
add VersionControlSettings file

Commit:
fix: washed up gif colors

Commit:
avatar AO + environment fresnel tweaks

Commit:
gif washed out fix 2

Commit:
revert fix

Commit:
UI atlas update

Commit:
remove SSS properties from toon shader

Commit:
fix: error handler hook not working + localhost debugging not working

Commit:
disable rim lighting for toon shader

Commit:
add more subtlety to AO toon shader visuals

Commit:
apply PR feedback

Commit:
fix: kernel error handling issue

- the error handler override was throwing an error, had to extract a custom unminified preprocessed version of the loader from the Unity build tools source code. Took the opportunity to also extract the debug symbols support (albeit it seems to not fully work yet)

- HUDs were not initialized when running a local build using make watch. Turns out we weren't waiting for EngineStarted() properly. Fixed it.

- Some tweaks to .gitignore

- Tweaks to MakeFile and publish.sh to avoid having old Unity files in the NPM package

- Bump Configuration.cs version to 1.0

Commit:
re-apply big sur fix

Commit:
re-apply drawing buffer fix

Commit:
fix: enable passing of baseUrl to loader instead of having it hardcoded in index.html

Commit:
Merge branch 'master' into chore/unity-2020-3-0-f1

Commit:
add SSAO setting

Commit:
add context menu to debug bridge methods for testing

Commit:
fix: float setting values broke presets spinbox

 - when the slider wholeNumbers property is true on a float setting, the flow ends up clamping the value and thus Equals call fails unexpectedly. Changed it so the wholeNumbers setting is set only if the setting is an int number.

Commit:
preset tweaks

Commit:
merge with master

Commit:
fix preset

Commit:
rename detailObjectCullingThreshold variable

Commit:
settings ordering and naming tweaks

Commit:
Merge branch 'master' into feat/add-ssao-settings

Commit:
shadows tentative fix

sandrade-dcl commented on unity-client/Assets/Scripts/MainScripts/DCL/Controllers/HUD/SettingsPanelHUD/Scripts/ControlsModule/SliderControlModel.cs:
```diff
@@ -15,9 +15,6 @@ public class SliderControlModel : SettingsControlModel
         [Tooltip("Max allowed value for the slider.")]
         public float sliderMaxValue;
 
-        [Tooltip("True if the slider values will be integers.")]
-        public bool sliderWholeNumbers;
```
A consequence of remove this param is that the sliders that had using it, now will show their values as decimal, like en these examples:

![image](https://user-images.githubusercontent.com/64659061/114210038-9a069e80-995f-11eb-8ee3-5bbd7989885a.png)

We should fix it.

pravusjif reviewed (approved):
I'm approving the PR taking into account that Santi's reported visual bug + objects culling setting will be fixed (as the culling is really needed for performance, affecting mostly the unneeded shadows rendering as you already know).

SSAO setting worked flawlessly for me.

Commit:
fix: whole numbers issue in sliders that needed it

- re-add whole numbers, but now it only affects the number view

Commit:
chore: add some utility fields to WSSController

Commit:
fix: shadows disabling emission

forced _EMISSION keyword on our lit shader so it doesn't get stripped away when we deactivate shadows

Commit:
Merge branch 'master' into feat/add-ssao-settings
